tests/test_trigger_cooldown.py

```diff
@@ -0,0 +1,75 @@
+import asyncio
+
+import pytest
+
+from cooldowns import (
+    CooldownBucket,
+    TriggerCooldown,
+    get_cooldown
+)
+from cooldowns.exceptions import CallableOnCooldown
+
+
+@pytest.mark.asyncio
+async def test_trigger_cooldown():
+    my_trigger_cooldown = TriggerCooldown(1, 0.3, CooldownBucket.all)
+
+    @my_trigger_cooldown
+    async def test_1(trigger_test = False):
+        if trigger_test:
+            await my_trigger_cooldown.trigger(20)
+        return 1
+
+    assert await test_1() == 1
+
+    with pytest.raises(CallableOnCooldown):
+        await test_1()
+
+    await asyncio.sleep(0.4)
+
+    assert await test_1(trigger_test = True)
+
+    await asyncio.sleep(0.4)
+    with pytest.raises(CallableOnCooldown):
+        await test_1()
+
+@pytest.mark.asyncio
+async def test_shared_trigger_cooldown():
+    my_shared_trigger_cooldown = TriggerCooldown(1, 0.3, CooldownBucket.all)
+
+    @my_shared_trigger_cooldown
+    async def test_1(*args, **kwargs):
+        return 1
+
+    @my_shared_trigger_cooldown
+    async def test_2(*args, **kwargs):
+        return 2
+
+    assert await test_1() == 1
+    assert await test_2() == 2
```

Comment:
Would this not trigger the cooldown? It's two calls to a cooldown which is 1/0.3 and I doubt it takes longer then 0.3 to execute both

Reply:
Oh, you are totally right. My apologies for the inconvenience. I had to go on a trip and I did it too fast. Second assert has to be removed.